torch/_inductor/lowering.py

```diff
@@ -1064,6 +1066,87 @@ def inner_fn(idx):
     )
 
 
+def _permute_to_axis_zero(x, axis):
+    new_axis_list = list(range(len(x.get_size())))
+    new_axis_list[axis] = 0
+    new_axis_list[0] = axis
+    y = permute(x, tuple(new_axis_list))
+    return y, new_axis_list
+
+
+@register_lowering(quantized_decomposed.quantize_per_channel, type_promotion_kind=None)
+def quantized_decomposed_quantize_per_channel(
+    input: TensorBox,
+    scales: TensorBox,
+    zero_points: TensorBox,
+    axis: int,
+    quant_min: int,
+    quant_max: int,
+    dtype: torch.dtype,
+) -> TensorBox:
+    assert len(scales.get_size()) == 1, "expect scales 1 dim"
+    assert len(zero_points.get_size()) == 1, "expect zero_points 1 dim"
+
+    if input.get_dtype() == torch.bfloat16:
+        input = input.to(torch.float32)
+    assert (
+        input.get_dtype() == torch.float32
+    ), f"Expecting input to have dtype torch.float32, but got dtype: {input.get_dtype()}"
+    assert axis < len(
+        input.get_size()
+    ), f"Expecting axis to be < {len(input.get_size())}"
+
+    input, permute_axis_list = _permute_to_axis_zero(input, axis)
+    original_input_size = input.get_size()
+    input = view(input, (original_input_size[0], -1))
+    res = maximum(
+        quant_min,
+        minimum(
+            quant_max,
+            add(
+                round(mul(input, unsqueeze(div(1.0, scales), 1))),
+                unsqueeze(zero_points, 1),
+            ),
+        ),
+    )
+    res = view(res, original_input_size)
+    out = permute(res, tuple(permute_axis_list))
+    return to_dtype(out, dtype)
```

Comment:
NIT: I think this would be more readable as a manually constructed `inner_fn`, where all this permuting and unsqueezing isn't necessary:
```
def inner_fn(idx):
    channel_idx = (idx[axis],)

    zero_point = zero_points_loader(channel_idx)
    inv_scale = ops.reciprical(scales_loader(channel_idx))

    val = ops.round(input_loader(idx) * inv_scale) + zero_point
    qmin, qmax = _create_constants(quant_min, quant_max, torch.float32)
    clamped = ops.max(qmin, ops.min(qmax, val))
    return ops.to_dtype(clamped, dtype)
   
```

Same principle applies to the dequantize operator.

Reply:
Thanks for the suggestion and changed, @peterbell10 please help to take a look again.